src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,438 @@
+        try {
+            let command = `az containerapp env create -n ${name} -g ${resourceGroup}`;
+            if (!util.isNullOrEmpty(location)) {
+                command += ` -l ${location}`;
+            }
+            await cpExec(`${command}`);
+        } catch (err) {
+            core.error(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Disables ingress on an existing Container App.
+     * @param name - the name of the Container App
+     * @param resourceGroup - the resource group that the Container App is found in
+     */
+    public async disableContainerAppIngress(name: string, resourceGroup: string) {
+        core.debug(`Attempting to disable ingress for Container App with name "${name}" in resource group "${resourceGroup}"`);
+        try {
+            const command = `az containerapp ingress disable -n ${name} -g ${resourceGroup}`;
+            await cpExec(`${command}`);
+        } catch (err) {
+            core.error(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Updates the ACR details on an existing Container App.
+     * @param name - the name of the Container App
+     * @param resourceGroup - the resource group that the Container App is found in
+     * @param acrName - the name of the Azure Container Registry (without the .azurecr.io suffix)
+     * @param acrUsername - the username used to authenticate with the Azure Container Registry
+     * @param acrPassword - the password used to authenticate with the Azure Container Registry
+     */
+    public async updateContainerAppRegistryDetails(name: string, resourceGroup: string, acrName: string, acrUsername: string, acrPassword: string) {
+        core.debug(`Attempting to set the ACR details for Container App with name "${name}" in resource group "${resourceGroup}"`);
+        try {
+            const command = `az containerapp registry set -n ${name} -g ${resourceGroup} --server ${acrName}.azurecr.io --username ${acrUsername} --password ${acrPassword}`;
+            await cpExec(`${command}`);
+        } catch (err) {
+            core.error(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Using the Oryx++ Builder, creates a runnable application image from the provided application source.
+     * @param imageToDeploy - the name of the runnable application image that is created and can be later deployed
+     * @param appSourcePath - the path to the application source on the machine
+     * @param runtimeStack - the runtime stack to use in the image layer that runs the application
+     */
+    public async createRunnableAppImage(
+        imageToDeploy: string,
+        appSourcePath: string,
+        runtimeStack: string) {
+        core.debug(`Attempting to create a runnable application image using the Oryx++ Builder with image name "${imageToDeploy}"`);
+        try {
+            let telemetryArg = `--env "CALLER_ID=azure-pipelines-rc-v1"`;
```

Comment:
TODO: What should the `telemetryArg` value be changed to?

Reply:
[`github-actions-v1`](https://github.com/Azure/container-apps-deploy-action/blob/main/action.yml#L495)

Reply:
Updated.